docs/docs/reference/production-deployment.md

```diff
@@ -0,0 +1,97 @@
+---
+title: Production Deployment
+description: >-
+  This article covers production deployment requirements and concerns
+---
+
+# Production Deployment
+
+This page covers the topic of running Pomerium in a production configuration.  See the [quick start section](../quick-start/) for canned example configurations.  
+
+Please also see [architecture](../#architecture) for information on component interactions.
+
+## Service Mode
+
+### All in One
+
+In smaller deployments or while testing, it may be desirable to run in "all-in-one" mode.  This reduces resource footprint and simplifies DNS configuration.  All URLs point at the same pomerium service instance.
+
+### Dedicated Services
+
+In larger footprints, it is recommended that you run Pomerium as a collection of independent services.  This limits blast radius in the event of vulnerabilities and allows for independing [scaling](#scaling) and monitoring of services.
+
+In more complicated deployments, you may also choose to use a single Authorize and/or Authenticate cluster to service multiple proxy clusters.
+
+## Scaling
+
+In dedicated service mode, you have the opportunity to scale the components of Pomerium independently.
+
+Pomerium's components are stateless, and may all be scaled horizontally or vertically.  In general, horizontal scaling is recommended.
+
+### Proxy
+
+Proxy service, as the name implies, is responsible for proxying all user traffic, in addition to performing Authentication and Authorization checks.  The proxy is directly in path for user traffic.
+
+Proxy will need resources scaled in conjunction with request count and may need average request size accounted for.  The heavier your user traffic, the more resources the Proxy service should have provisioned.
+
+### Authorize
+
+The Authorize serivce is responsible for policy checks during requests.  It is in the hot path for user requests but does not directly handle user traffic.
+
+Authorize will need resources scaled in conjunction with request count.  Request size and type should be of a constant complexity.  Policy complexity/length may have some impact on performance.
```

Comment:
Well said. In theory, authZ lookups are ~ O(1) so they should scale proportionally with requests minus whatever the gRPC overhead is.

Reply:
Ah, I misremembered the policy lookup logic.  Yes I agree, it looks very close to O(1).